File: GameLibrary/src/cc/lib/swing/AWTTransparencyFilter.java
package cc.lib.swing;

import java.awt.Color;
import java.awt.image.RGBImageFilter;

public class AWTTransparencyFilter extends RGBImageFilter {

	int targetColor;
	
	public AWTTransparencyFilter(Color color) {
		this(AWTUtils.colorToInt(color));
	}
	
	public AWTTransparencyFilter(int color) {
		targetColor = color & 0x00ffffff;
	}	
	
	public int filterRGB(int x, int y, int rgb) {
	    int c = (rgb & 0x00ffffff);
		if (c == targetColor) {
			return 0;
		}
		return rgb;
	}
	
}

File: SwingApps/src/cc/lib/swing/AWTTransparencyFilter.kt
package cc.lib.swing

import java.awt.Color
import java.awt.image.RGBImageFilter

class AWTTransparencyFilter(color: Int) : RGBImageFilter() {
	var targetColor: Int

	constructor(color: Color) : this(AWTUtils.colorToInt(color)) {}

	init {
		targetColor = color and 0x00ffffff
	}

	override fun filterRGB(x: Int, y: Int, rgb: Int): Int {
		val c = rgb and 0x00ffffff
		return if (c == targetColor) {
			0
		} else rgb
	}
}